addrmgr/knownaddress.go

```diff
@@ -95,7 +95,7 @@ func (ka *KnownAddress) isBad() bool {
 	}
 
 	// Never succeeded?
-	if ka.lastsuccess.IsZero() && ka.attempts >= numRetries {
+	if ka.lastsuccess.IsZero() || ka.attempts >= numRetries {
```

Comment:
Will need to review all the surrounding code to be sure, but this doesn't look right to me at first glance.  An address should not be marked bad before it's had a few tries (`numRetries` in this case) to succeed.

With this change, it seems like an address will be marked bad when `lastsuccess` is zero, which should always happen before the first connection has been made.